modules/tactical/src/backend.ts

```diff
@@ -65,12 +103,45 @@ export class FakeBackend implements Backend {
   load(object: VersionedObject, broadcast: boolean = true): void {
     this.objects[serializeValue(object.key)] = object;
     if (broadcast) {
-      this.subject.onNext(object);
+      this.dataSubject.onNext(object);
     }
   }
 
+  mutate(key: Object, value: Object, baseVersion: string, mutationId: Object): void {
+    var keyStr = serializeValue(key);
+    // Check for version compatibility.
+    if (this.objects.hasOwnProperty(keyStr)) {
+      var old = this.objects[keyStr];
+      if (old.version !== baseVersion) {
+        // Newer version is available. Reject mutation by returning new version.
+        this.dataSubject.onNext(old);
+        return;
+      }
+    }
+    if (mutationId.hasOwnProperty('fail') && mutationId['fail']) {
```

Comment:
As it stands, this is very confusing to me. Why would the DM call mutate and provide a mutationId that contains a 'fail' property?